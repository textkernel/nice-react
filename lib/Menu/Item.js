import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Item = ({ children, className, current, onClick, label, href, target, disabled, icon }) => (<li className={ classNames({
    disabled,
    current
}) }>
    <a href={ href } onClick={ onClick } target={ target } className={ className }>
        { label }
    </a>
    {
        children ?
        <ul>
            { children }
        </ul> : null
    }
</li>);

Item.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.node,
        PropTypes.array
    ]),
    current: PropTypes.bool,
    onClick: PropTypes.func,
    label: PropTypes.string,
    href: PropTypes.string,
    target: PropTypes.string,
    disabled: PropTypes.bool,
    className: PropTypes.string
};

Item.defaultProps = {
    children: null,
    current: false,
    onClick: null,
    label: '',
    href: '#',
    target: null,
    disabled: false,
    className: null 
};

export default Item;
