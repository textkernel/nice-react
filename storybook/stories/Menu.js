import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobsOptions, withKnobs, boolean, number, select } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';

import { Menu } from 'nice-react';

const stories = storiesOf('Menu', module);

stories.addDecorator((story, context) => withInfo('')(story)(context));
stories.addDecorator(withKnobs);

const onClick = e => {
    e.preventDefault();
};

stories.add('Default (horizontal)', () => (
    <Menu>
        <Menu.Item className="icon fa-home" label="Home" onClick={ onClick } current />
        <Menu.Item className="icon fa-map-pin" label="Cities" onClick={ onClick } />
        <Menu.Item label="Regions" onClick={ onClick } />
        <Menu.Item className="icon fa-globe" label="Countries" onClick={ onClick } disabled />
    </Menu>
));

stories.add('Stacked & Narrow', () => (
    <Menu stacked narrow icons>
        <Menu.Item className="icon fa-home" label="Home" onClick={ onClick } current />
        <Menu.Item className="icon fa-map-pin" onClick={ onClick } label="Cities">
            <Menu.Item label="Amsterdam" onClick={ onClick } />
            <Menu.Item label="Paris" onClick={ onClick } />
            <Menu.Item label="Berlin" onClick={ onClick } disabled />
            <Menu.Item label="Rome" onClick={ onClick } />
            <Menu.Item label="Madrid" onClick={ onClick } />
        </Menu.Item>
        <Menu.Item label="Regions" onClick={ onClick } />
        <Menu.Item className="icon fa-globe" label="Countries" disabled onClick={ onClick } />
    </Menu>
));

stories.add('Custom properties', () => {
    const stacked = boolean('Stacked', false);
    const narrow = boolean('Narrow', false);
    const icons = stacked && boolean('Has icons (only if stacked)', true);
    const items = number('Number of items', 5);
    const currentItem = number('Current item', 1);

    return (<Menu stacked={ stacked } narrow={ narrow } icons={ icons }>
        {
            Array(items).fill(0).map((v, k) => (
                <Menu.Item current={ currentItem && k + 1 === currentItem } onClick={ onClick } label={ `Menu item ${ k + 1 }` } />
            ))
        }
    </Menu>);
});
