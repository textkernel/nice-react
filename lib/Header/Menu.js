import React from 'react';
import PropTypes from 'prop-types';
import Item from './Item';
import classNames from 'classnames';

const Menu = ({ children, alternate, level }) => (
    <ul className={ classNames('nice-header-menu', {
        'menu-alt': alternate
    }) }>
        { React.Children.map(children, child => (
            React.cloneElement(child, { level })
        )) }
    </ul>
);

Menu.propTypes = {
    alternate: PropTypes.bool,
    level: PropTypes.number
};

Menu.defaultProps = {
    alternate: false,
    level: 0
};

Menu.Item = Item;

export default Menu;
