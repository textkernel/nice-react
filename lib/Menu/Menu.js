import React from 'react';
import PropTypes from 'prop-types';
import props from '../props';
import Item from './Item';

const Menu = ({ children, icons, stacked, narrow }) => {
    const classes = ['nice-menu'];
    if (icons) {
        classes.push('has-icons');
    }
    if (stacked) {
        classes.push('stacked');
    }
    if (narrow) {
        classes.push('narrow');
    }

    return (<ul className={ classes.join(' ') }>
        { children }
    </ul>);
};

Menu.Item = Item;

Menu.propTypes = {
    children: PropTypes.node,
    icons: PropTypes.bool,
    stacked: PropTypes.bool,
    narrow: PropTypes.bool
};

Menu.defaultProps = {
    children: null,
    icons: false,
    stacked: false,
    narrow: false
};

export default Menu;