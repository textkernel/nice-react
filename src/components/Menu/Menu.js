import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Item from './Item';
import './Menu.scss';

const Menu = ({ className, style, children, icons, stacked, narrow }) => (
    <ul
        className={ classnames('nice-menu', className, {
            'has-icons': icons,
            narrow,
            stacked
        }) }
        style={style}
    >
        { children }
    </ul>
);

Menu.Item = Item;

Menu.propTypes = {
    className: PropTypes.string,
    children: PropTypes.node,
    icons: PropTypes.bool,
    stacked: PropTypes.bool,
    style: PropTypes.object,
    narrow: PropTypes.bool
};

Menu.defaultProps = {
    className: null,
    children: null,
    icons: false,
    stacked: false,
    style: {},
    narrow: false
};

export default Menu;
