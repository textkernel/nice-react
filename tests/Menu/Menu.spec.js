import React from 'react';
import Menu from '../../lib/Menu';

const onClick = () => {};

describe('<Menu />', () => {
    it('should render default (horizontal) Menu', () => {
        const wrapper = shallow(<Menu>
        <Menu.Item className="icon fa-home" label="Home" onClick={ onClick } current />
        <Menu.Item className="icon fa-map-pin" label="Cities" onClick={ onClick } />
        <Menu.Item label="Regions" onClick={ onClick } />
        <Menu.Item className="icon fa-globe" label="Countries" onClick={ onClick } disabled />
    </Menu>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render stacked / narrow menu', () => {
        const wrapper = shallow(<Menu stacked narrow icons>
        <Menu.Item className="icon fa-home" label="Home" onClick={ onClick } current />
        <Menu.Item className="icon fa-map-pin" onClick={ onClick } label="Cities">
            <Menu.Item label="Amsterdam" onClick={ onClick } />
            <Menu.Item label="Paris" onClick={ onClick } />
            <Menu.Item label="Berlin" onClick={ onClick } disabled />
            <Menu.Item label="Rome" onClick={ onClick } />
            <Menu.Item label="Madrid" onClick={ onClick } />
        </Menu.Item>
        <Menu.Item label="Regions" onClick={ onClick } />
        <Menu.Item className="icon fa-globe" label="Countries" disabled onClick={ onClick } />
    </Menu>);
        expect(wrapper).toMatchSnapshot();
    });
});
