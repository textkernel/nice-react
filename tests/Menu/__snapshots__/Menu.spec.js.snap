// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render default (horizontal) Menu 1`] = `
<ul
  className="nice-menu"
>
  <Item
    className="icon fa-home"
    current={true}
    disabled={false}
    href="#"
    label="Home"
    onClick={[Function]}
    target={null}
  />
  <Item
    className="icon fa-map-pin"
    current={false}
    disabled={false}
    href="#"
    label="Cities"
    onClick={[Function]}
    target={null}
  />
  <Item
    className={null}
    current={false}
    disabled={false}
    href="#"
    label="Regions"
    onClick={[Function]}
    target={null}
  />
  <Item
    className="icon fa-globe"
    current={false}
    disabled={true}
    href="#"
    label="Countries"
    onClick={[Function]}
    target={null}
  />
</ul>
`;

exports[`<Menu /> should render stacked / narrow menu 1`] = `
<ul
  className="nice-menu has-icons stacked narrow"
>
  <Item
    className="icon fa-home"
    current={true}
    disabled={false}
    href="#"
    label="Home"
    onClick={[Function]}
    target={null}
  />
  <Item
    className="icon fa-map-pin"
    current={false}
    disabled={false}
    href="#"
    label="Cities"
    onClick={[Function]}
    target={null}
  >
    <Item
      className={null}
      current={false}
      disabled={false}
      href="#"
      label="Amsterdam"
      onClick={[Function]}
      target={null}
    />
    <Item
      className={null}
      current={false}
      disabled={false}
      href="#"
      label="Paris"
      onClick={[Function]}
      target={null}
    />
    <Item
      className={null}
      current={false}
      disabled={true}
      href="#"
      label="Berlin"
      onClick={[Function]}
      target={null}
    />
    <Item
      className={null}
      current={false}
      disabled={false}
      href="#"
      label="Rome"
      onClick={[Function]}
      target={null}
    />
    <Item
      className={null}
      current={false}
      disabled={false}
      href="#"
      label="Madrid"
      onClick={[Function]}
      target={null}
    />
  </Item>
  <Item
    className={null}
    current={false}
    disabled={false}
    href="#"
    label="Regions"
    onClick={[Function]}
    target={null}
  />
  <Item
    className="icon fa-globe"
    current={false}
    disabled={true}
    href="#"
    label="Countries"
    onClick={[Function]}
    target={null}
  />
</ul>
`;
